From c8e2a99e68109b386c541dcc157f1f64ca8cbd17 Mon Sep 17 00:00:00 2001
From: xqzr <34030394+xqzr@users.noreply.github.com>
Date: Fri, 17 Dec 2021 00:25:16 +0800
Subject: [PATCH] fix unsupported cipher method: xChaCha20-IETF-Poly1305

---
 infra/conf/shadowsocks.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/infra/conf/shadowsocks.go
+++ b/infra/conf/shadowsocks.go
@@ -59,7 +59,7 @@ func (v *ShadowsocksServerConfig) Build(
 				return nil, newError("Shadowsocks password is not specified.")
 			}
 			if account.CipherType < shadowsocks.CipherType_AES_128_GCM ||
-				account.CipherType > shadowsocks.CipherType_CHACHA20_POLY1305 {
+				account.CipherType > shadowsocks.CipherType_XCHACHA20_POLY1305 {
 				return nil, newError("unsupported cipher method: ", user.Cipher)
 			}
 			config.Users = append(config.Users, &protocol.User{
